pkg/engines/c/engine.go

```diff
@@ -0,0 +1,112 @@
+package C
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/kampanosg/lazytest/pkg/engines"
+	"github.com/kampanosg/lazytest/pkg/models"
+)
+
+const icon = "C"
+
+type cNode struct {
+	Ref      any
+	Name     string
+	Children map[string]*cNode
+}
+
+type CEngine struct {
+	Runner engines.Runner
+}
+
+func NewCEngine(r engines.Runner) *CEngine {
+	return &CEngine{
+		Runner: r,
+	}
+}
+
+func (c *CEngine) GetIcon() string { return icon }
+
+func (c *CEngine) Load(dir string) (*models.LazyTree, error) {
+	// should make the command configurable. maybe some thing like https://docs.helix-editor.com/languages.html
+	// Testing in C normally always uses custom binaries associated with the make file
+	// could extrapolate that to have a generic configurable engine perhaps!
+	o, err := c.Runner.RunCmd(fmt.Sprintf("cd %s && make test TFLAGS=--list", dir))
```

Comment:
I should make the command configurable. maybe some thing like https://docs.helix-editor.com/languages.html.
Testing in C normally always uses custom binaries associated with the make file
you. Could extrapolate that to have a generic configurable engine perhaps!

Let me know if you like this idea.